### ISSUE TITLE
Better handling of regional analyses with single destination layer, cutoff, or percentile

### ISSUE THREAD

**Issue by ansoncfit** [conveyal/analysis-ui#1595]
Legacy regional analyses have `grid`, `travelTimePercentile`, and `cutoffMinutes` properties. R5 v5.10 added corresponding array properties `destinationPointSetIds`, `travelTimePercentiles`, and `cutoffsMinutes`. Multi-cutoff and multi-percentile regional analyses will have these latter two arrays populated, and the legacy `travelTimePercentile` and `cutoffMinutes` properties set to a special value (-2). Multi-destination layer analyses have `grid` set to `null`.

The previous UI release handled both cases, showing dropdown selectors only if the newer arrays were present (see https://github.com/conveyal/analysis-ui/blob/f3a5b12e86464125358bd984525803ba25ca8ca6/lib/components/analysis/regional.tsx#L210-L238), and using the legend to show destination layer name, cutoff, and percentile in both cases.

The current dev UI shows selectors regardless; they are empty for legacy/single-value analyses (thanks to https://github.com/conveyal/analysis-ui/blob/12c62421dfe376c471303d85b302427cdcf2a17f/lib/regional/components/variant-selectors.tsx#L27-L28). Furthermore, without a selector the destination layer name is not visible (it's not in the legend).

**Comment by trevorgerhardt** [comment]
We are already running a database migration for this release. Should we add one more and convert all old regional analyses to the new style?

**Comment by abyrd** [comment]
> We are already running a database migration for this release. Should we add one more and convert all old regional analyses to the new style?

Yes it does seem like a database migration might be appropriate here. Considering that the UI code has already been simplified to no longer handle the legacy style, and considering that we are already doing a significant database migration this time around. Are we sure the database migration wouldn't have repercussions with any other code (backend or worker)?

**Comment by abyrd** [comment]
On further investigation the migration will be problematic. The backend uses the presence or absence of the plural field names (around RegionalAnalysisController L283) to format the filename of the original multi-cutoff results to open. If all database records are migrated to plural array-typed cutoffs/percentiles/destinations fields it will always look for `"%s_%s_P%d.access"`. There are at least four possibilities:
- leave the old singular fields in the database and Java model objects and put special cases in the UI (status quo)
- migrate the files on S3 to all use new-format names (so far we've avoided s3 migrations)
- add another field to the database and Java objects flagging old formats (this is not really less ugly than just retaining the old singular fields)
- try the filename formats from newest to oldest, checking whether each exists on S3 before falling back to older ones, and checking that arrays are of length 1 for the older formats

I would probably go with the first or last option (status quo or automated fallback to older file format names).

**Comment by abyrd** [comment]
Proposed fallback mechanism for use after database migration: https://github.com/conveyal/r5/pull/749
